linode/manager
# upcoming: [DI-20050] - Added Data Rollup logic for graph data

## Description 📝
Added data rollup logic to convert the graph data  to highest possible unit based on the base unit of data we receive from widget configuration.

## Changes  🔄
List any change relevant to the reviewer.
1. Added a unit conversion utility file to handle data rollup & conversion logic
2. Modified utils.ts to add a logic to generate label name for graph.
3. Modified JWE token payload from resource_id to resource_ids based on api specs
4. Updated position of message for empty data in the graph.

## Target release date 🗓️
15th August 2024

## Preview 📷
**Include a screenshot or screen recording of the change**

:bulb: Use `<video src="" />` tag when including recordings in table.

| Before  | After   |
| ------- | ------- |
|![Screenshot 2024-08-05 at 7 49 36 PM](https://github.com/user-attachments/assets/96aeb377-09e6-4e48-89fa-8cdd36eeac10)|![Screenshot 2024-08-05 at 7 37 01 PM](https://github.com/user-attachments/assets/cf1659ae-e1a2-4b8b-8197-c93bd35dd58a)|
|![Screenshot 2024-08-05 at 7 37 46 PM](https://github.com/user-attachments/assets/84663504-8133-4645-a672-75f100522d9c)|![Screenshot 2024-08-05 at 7 36 52 PM](https://github.com/user-attachments/assets/074c8121-9fb1-4bd0-b6c2-b479f32e5a29)|


## How to test 🧪

1. Enable mock services as some endpoints are not ready to use
2. As there is not much visual change in the UI you can simulate the unit conversion logic by passing different values & units

Note: You may notice some flickering because of continuous re-rendering but that will be resolved once global filter change handler will be implemented.



*Check all that apply*
